### PR TITLE
Update win_service.py

### PR DESCRIPTION
**Pull request by jlayze**
### What does this PR do?
Added named variables `start_type` and `start_delayed` in win_service.enable.  This allows for setting Manual and Auto (Delayed) start types from states.service.running and states.service.dead.

Example state.sls:
```
"Patch Agent":
  service.running:
    - enable: True
    - start_type: manual
```
Or
```
"Patch Agent":
  service.dead:
    - enable: True
    - start_delayed: True
```
### What issues does this PR fix or reference?
Set StartType for Windows Service from states.service module #42052

### Tests written?
Yes, and I found if the service is running, then changing from Auto to Auto(Delayed) or vice versa doesn't work due to the before_toggle_enable_status check in states.service.running:
```
before_toggle_status = __salt__['service.status'](name, sig)
    if 'service.enabled' in __salt__:
        before_toggle_enable_status = __salt__['service.enabled'](name)
    else:
        before_toggle_enable_status = True

    # See if the service is already running
    if before_toggle_status:
        ret['comment'] = 'The service {0} is already running'.format(name)
        if enable is True and not before_toggle_enable_status:
            ret.update(_enable(name, None, **kwargs))
        elif enable is False and before_toggle_enable_status:
            ret.update(_disable(name, None, **kwargs))
        return ret
```
Also found the same to be true with states.service.dead if the service is already stopped.

### CODE REVIEW

File: salt/modules/win_service.py [hunk 1]
```diff
@@ -817,6 +817,7 @@ def enable(name, **kwargs):
 
     Args:
         name (str): The name of the service to enable.
+        **kwargs supports StartType
```
Comment (twangboy) [inline salt/modules/win_service.py]: This should probably add two named arguments here. `start_type` with a default of `auto` and `start_delayed` with a default of `False`. To conform with other parameter names in salt please avoid CamelCase.

There needs to be additional documentation here as well for both parameters... similar to that for `modify`.

File: salt/modules/win_service.py [hunk 2]
```diff
@@ -827,8 +828,20 @@ def enable(name, **kwargs):
...
+                return info(name)['StartType'] == 'Manual'
+            elif key == 'StartType' and value == 'Delayed':
+                modify(name, start_type='Auto', start_delayed=True)
+                return info(name)['StartTypeDelayed']
+            else:
+                log.error('Invalid argument for win_service.enable!')
+                return False
+    else:
+        modify(name, start_type='Auto')
+        return info(name)['StartType'] == 'Auto'
```
Comment (twangboy) [inline salt/modules/win_service.py]: If you add the two parameters above, you don't need the whole `if kwargs is not None:` block. Something like this should work.

```
modify(name, start_type=start_type, start_delayed=start_delayed)
status = info(name)
return statys['StartType'].lower() == start_type.lower() and status['StartTypeDelayed'] == start_delayed
```

**Review by twangboy — CHANGES_REQUESTED** [review]

**Comment by jlayze** [comment]
@twangboy Thanks, this looks much cleaner now.

I still needed to keep an IF block for the return, as the 'Delayed' flag isn't cleared when changing from Auto(Delayed) to Manual or Disabled, so service.enable was reporting a failure to enable start at boot when changing between Manual and Disabled after previously being set to Auto(Delayed).

**Review by twangboy — APPROVED** [review]

**Comment by twangboy** [comment]
@jlayze Looks like we got some failing tests...

**Comment by jlayze** [comment]
@twangboy aren't the Ubuntu and CentOS failures expected for a Windows module?  win_service.py should never be used on anything but Windows, and it's working as expected in all the tests I've run.

**Comment by twangboy** [comment]
@jlayze Yeah, even though it's a Windows only function, some unit tests still run on Linux. Return values are Mocked so they still run. It looks like the `return_value` for the `MagicMock` needs to have an item added to the dictionary for `StartTypeDelayed`.

So, it's in the file `tests/unit/modules/test_win_service.py` on line `216` or so (in the `test_enable` function).
It needs to be something like this:
```
        mock_info = MagicMock(return_value={'StartType': 'Auto',
                                            'StartTypeDelayed': False})
```

**Comment by jlayze** [comment]
@twangboy created PR #42091 to update test_win_service.py

**Comment by twangboy** [comment]
@jlayze You need to modify `test_win_service.py` in this PR so that the test runs against the new changes in `win_service.py`.

**Comment by jlayze** [comment]
@twangboy So I've made the required change to `test_win_service.py` and now `test_enable` is passing, but there are 31 other failures that are new.  I'm not sure where the 2 new checks and these errors came from.  I tried updating this PR's branch with the current `development` branch, but that hasn't helped.

**Comment by twangboy** [comment]
@jlayze These test failures are unlrelated to this PR. Great job and thanks for your submission.

**Review by rallytime — APPROVED** [review]

**Pull request merged**